<launch>

<arg name="map_size_x" default="10.0"/>
<arg name="map_size_y" default="10.0"/>
<arg name="map_size_z" default="10.0"/>
<arg name="dynamic_barrier" default="0"/>
<arg name="arg_is_track" default = "0"/>
<arg name="start_x" default=" 0.0"/>
<arg name="start_y" default=" 0.0"/>
<arg name="start_z" default=" 2"/>

  <!-- Start rviz -->
  <node name="rviz" pkg="rviz" type="rviz" output="screen"
       args="-d $(find grid_path_searcher)/launch/rviz_config/demo.rviz"> 
  </node>

  <node pkg="grid_path_searcher" type="hybridAstar" name="hybridAstar" output="screen" required = "true">
      <remap from="~waypoints"       to="/waypoint_generator/waypoints"/>
      <remap from="~map"             to="/random_complex/global_map"/> 

      <param name="map/margin"       value="0.0" />
      <param name="map/resolution"   value="0.1" />
      <param name="map/x_size"       value="$(arg map_size_x)"/>
      <param name="map/y_size"       value="$(arg map_size_y)"/>
      <param name="map/z_size"       value="$(arg map_size_z)"/>
      <param name="is_dynamic"       value="$(arg dynamic_barrier)"/>
      <param name="is_track"         value="$(arg arg_is_track)"/>
      <param name="vis/vis_traj_width" value="0.05" /> 
      <param name="planning/start_x" value="$(arg start_x)"/>
      <param name="planning/start_y" value="$(arg start_y)"/>
      <param name="planning/start_z" value="$(arg start_z)"/>
      
      <param name="search/max_tau"  value = "0.25"/>
      <param name="search/init_max_tau" value = "0.8"/>
      <param name="search/max_vel"  value = "100000000000.0" />
      <param name="search/max_acc"  value = "4.0"/>
      <param name="search/w_time"  value = "10.0"/>
      <param name="search/lambda_heu" value = "5"/>
      <param name="search/check_num"  value = "5"/>     


  </node>

  <node pkg="drone_node" type="drone_node" name="drone_node" output="screen" required = "true">
      <remap from="~waypoints"       to="/waypoint_generator/waypoints"/>
      <remap from="~map"             to="/random_complex/global_map"/> 

      <param name="map/margin"       value="0.0" />
      <param name="map/resolution"   value="0.1" />
      <param name="map/x_size"       value="$(arg map_size_x)"/>
      <param name="map/y_size"       value="$(arg map_size_y)"/>
      <param name="map/z_size"       value="$(arg map_size_z)"/>

      <param name="planning/start_x" value="$(arg start_x)"/>
      <param name="planning/start_y" value="$(arg start_y)"/>
      <param name="planning/start_z" value="$(arg start_z)"/>
  </node>


  <node pkg ="grid_path_searcher" name ="random_complex" type ="random_complex" output = "screen">    
    
      <param name="init_state_x"   value="$(arg start_x)"/>
      <param name="init_state_y"   value="$(arg start_y)"/>
      <param name="init_dynamic"   value="$(arg dynamic_barrier)"/>

      <param name="map/x_size"     value="$(arg map_size_x)" />
      <param name="map/y_size"     value="$(arg map_size_y)" />
      <param name="map/z_size"     value="$(arg map_size_z)" />

      <param name="map/circle_num" value="0"/>        
      <param name="map/obs_num"    value="50"/>        
      <param name="map/resolution" value="0.1"/>
              

      <param name="ObstacleShape/lower_rad" value="0.1"/>
      <param name="ObstacleShape/upper_rad" value="0.7"/>
      <param name="ObstacleShape/lower_hei" value="1.0"/>
      <param name="ObstacleShape/upper_hei" value="3.0"/>

      <param name="CircleShape/lower_circle_rad"   value="0.6"/>        
      <param name="CircleShape/upper_circle_rad"   value="2.0"/>        
      
      <param name="sensing/rate"   value="0.5"/>        
  </node>

  <node pkg="waypoint_generator" name="waypoint_generator" type="waypoint_generator" output="screen">
      <remap from="~goal" to="/goal"/>
      <param name="waypoint_type" value="manual-lonely-waypoint"/>    
  </node>
  
  

</launch>
